frontend/src/components/partnerMapswipeStats/contributionsHeatmap.js

```diff
@@ -0,0 +1,216 @@
+import React, { useEffect, useRef } from 'react';
+import mapboxgl from 'mapbox-gl';
+import { latLngToCell, cellToBoundary } from 'h3-js';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+import 'mapbox-gl/dist/mapbox-gl.css';
+
+import { MAPBOX_TOKEN, MAP_STYLE, CHART_COLOURS } from '../../config';
+import messages from './messages';
+import './contributionsHeatmap.css';
+
+mapboxgl.accessToken = MAPBOX_TOKEN;
+
+export const ContributionsHeatmap = ({ contributionsByGeo = [] }) => {
+  const mapContainer = useRef(null);
+  const map = useRef(null);
+
+  useEffect(() => {
+    if (map.current) return; // initialize map only once
+
+    map.current = new mapboxgl.Map({
+      container: mapContainer.current,
+      style: MAP_STYLE,
+      center: [0, 0],
+      zoom: 1.25,
+    });
+
+    map.current.scrollZoom.disable();
+    map.current.addControl(new mapboxgl.NavigationControl());
+
+    const getStyle = (row) => {
+      const styles = [
+        {
+          color: CHART_COLOURS.red,
+          opacity: 0.2,
+        },
+        {
+          color: CHART_COLOURS.red,
+          opacity: 0.3,
+        },
+        {
+          color: CHART_COLOURS.red,
+          opacity: 0.4,
+        },
+        {
+          color: CHART_COLOURS.red,
+          opacity: 0.6,
+        },
+        {
+          color: CHART_COLOURS.red,
+          opacity: 0.8,
+        },
+      ];
+
+      if (Number(row.totalContributionCount) === 0) {
+        return { opacity: 0 };
+      }
+
+      if (Number(row.totalContributionCount) < 250) {
+        return styles[0];
+      }
+      if (Number(row.totalContributionCount) < 500) {
+        return styles[1];
+      }
+      if (Number(row.totalContributionCount) < 1000) {
+        return styles[2];
+      }
+      if (Number(row.totalContributionCount) < 1500) {
+        return styles[3];
+      }
+      return styles[4];
+    };
+
+    const getHexagonFeatures = (res = 1) => {
+      const hexagonsArray = contributionsByGeo.map((data) => {
+        const h3Index = latLngToCell(data.geojson.coordinates[1], data.geojson.coordinates[0], res);
+        return {
+          hexindex7: h3Index,
+          totalContributionCount: data.totalContribution,
+        };
+      });
+
+      const features = hexagonsArray.map((row) => {
+        const style = getStyle(row);
+        return {
+          type: 'Feature',
+          properties: {
+            color: style.color,
+            opacity: style.opacity,
+            id: row.hexindex7,
+          },
+          geometry: {
+            type: 'Polygon',
+            coordinates: [cellToBoundary(row.hexindex7, true)],
+          },
+        };
+      });
+
+      return features;
+    };
+
+    const zoomToH3ResMapping = {
```

Comment:
can we write a function for these mappings?

Reply:
Not sure how a function can help here since we have pre-established static values.
We can retain it this way for now.